kolibri/plugins/hooks.py

```diff
@@ -1,59 +1,230 @@
 """
-Hooks are used for your plugin's ``hooks`` attribute, a dictionary
-with hooks you wish to use. The keys are pre-defined constants in this module or arbitrary strings. For instance::
-
-    from kolibri.plugins.base import KolibriPluginBase
-    from kolibri.plugins.hooks import NAVIGATION_POPULATE
-
-    class MyPlugin(KolibriPluginBase):
-
-        def my_navigation(self, navigation_items):
-            navigation_items.append(
-                {
-                    'menu_name': _("My list"),
-                    'menu_url': reverse_lazy("my_plugin:list_of_things")
-                }
-            )
-
-        def hooks(self):
-            return {
-                NAVIGATION_POPULATE: self.my_navigation
-            }
+Kolibri Hooks API
+-----------------
+
+What are hooks
+~~~~~~~~~~~~~~
+
+Hooks are classes that define *something* that happens. It means that you can
+"hook into a component" in Kolibri and have it do a predefined and
+parameterized *thing*. For instance, Kolibri could ask all its plugins who
+wants to add something to the user settings panel, and its then up to the
+plugins to inherit from that specific hook and feed back the parameters that
+the hook definition expects.
+
+The consequences of a hook callback are located anywhere in Kolibri. Each hook
```

Comment:
Absolutely! This will be a very common extension for plugins, and since it's only applied once and only always applied the same place, it makes sense to implement it in the plugin class IMO -- and not as a hook.